turbinia/workers/analysis/jenkins.py

```diff
@@ -0,0 +1,162 @@
+    """Run the Jenkins worker.
+
+    Args:
+        evidence (Evidence object):  The evidence to process
+        result (TurbiniaTaskResult): The object to place task results into.
+
+    Returns:
+        TurbiniaTaskResult object.
+    """
+    # What type of evidence we should output.
+    output_evidence = ReportText()
+
+    # Where to store the resulting output file.
+    output_file_name = 'jenkins_analysis.txt'
+    output_file_path = os.path.join(self.output_dir, output_file_name)
+
+    # Set the output file as the data source for the output evidence.
+    output_evidence.local_path = output_file_path
+
+    try:
+      collected_artifacts = extract_artifacts(
+        artifact_names=['JenkinsConfigFile'],
+        disk_path=evidence.local_path,
+        output_dir=os.path.join(self.output_dir, 'artifacts')
+      )
+    except TurbiniaException as e:
+      result.close(self, success=False, status=str(e))
+      return result
+
+    version = None
+    credentials = []
+    for filepath in collected_artifacts:
+        with open(filepath, 'r') as input_file:
+          config = input_file.read()
+
+        extracted_version = self._extract_jenkins_version(config)
+        extracted_credentials = self._extract_jenkins_credentials(config)
+
+        if extracted_version:
+          version = extracted_version
+
+        credentials.extend(extracted_credentials)
+
+    analysis_report = self.analyze_jenkins(version, credentials)
+    output_evidence.text_data = analysis_report
+
+    # Write the report to the output file.
+    with open(output_file_path, 'w') as fh:
+      fh.write(output_evidence.text_data.encode('utf8'))
+      fh.write('\n'.encode('utf8'))
+
+    # Add the resulting evidence to the result object.
+    result.add_evidence(output_evidence, evidence.config)
+    result.close(self, success=True)
+
+    return result
+
+  @staticmethod
+  def _extract_jenkins_version(config):
+    """Extract version from Jenkins configuration files.
+
+    Args:
+      config (str): configuration file content.
+
+    Returns:
+      str: The version of Jenkins.
+    """
+    version = None
+    version_re = re.compile('<version>(.*)</version>')
+    version_match = re.search(version_re, config)
+
+    if version_match:
+      version = version_match.group(1)
+
+    return version
+
+  @staticmethod
+  def _extract_jenkins_credentials(config):
+    """Extract credentials from Jenkins configuration files.
+
+    Args:
+      config (str): configuration file content.
+
+    Returns:
+      list: of tuples with username and password hash.
+    """
+    credentials = []
+    password_hash_re = re.compile('<passwordHash>#jbcrypt:(.*)</passwordHash>')
+    username_re = re.compile('<fullName>(.*)</fullName>')
+
+    password_hash_match = re.search(password_hash_re, config)
+    username_match = re.search(username_re, config)
+
+    if username_match and password_hash_match:
+      username = username_match.group(1)
```

Comment:
Seems like as it is this method will only ever return one tuple of user/hash.  Is it possible that there are multiple usernames/hashes in this file? If so, can we change this to extract all of them, and if not, should we change the method to just return a tuple?